docs/sql-functions/approximate-aggregate/approx-topk.md

```diff
@@ -0,0 +1,169 @@
+
+## What is `approx_topk`?
+The `approx_topk()` function returns an approximate list of the top K most frequently occurring values in a specified field. It uses the Space-Saving algorithm, a memory-efficient approach designed for high-cardinality data and distributed processing, providing significant [performance benefits](#performance-comparison). 
+
+> To find the top K values based on the number of distinct values in another field, use the [approx_topk_distinct() function](../approx-topk-distinct/).
+
+---
+
+## Query Syntax
+```sql
+
+SELECT approx_topk(field_name, K) FROM "stream_name"
+```
+Here:
+
+- `field_name`: The field for which top values should be retrieved.
+- `K`: The number of top values to return.
+- `stream_name`: The stream containing the data.
+
+**Example**
+```sql
+SELECT approx_topk(clientip, 10) FROM "demo1"
+```
+This query returns an approximate list of the `top k` most frequently occurring values in the `clientip` field from the `demo1` stream.
+
+**Result of `approx_topk`** <br>
+The result is returned as an array of objects, where each object includes the value and its corresponding count. For example:
+
+```json
+{
+  "item": [ { "value": "192.168.1.100", "count": 2650 }, { "value": "10.0.0.5", "count": 2230 }, { "value": "203.0.113.50", "count": 2210 }, { "value": "198.51.100.75", "count": 1979 }, { "value": "172.16.0.10", "count": 1939 } ]
+}
+```
+
+### Use `approx_topk` With `unnest`
+To convert these nested results into individual rows, use the `unnest()` function.
+
+```sql
+SELECT item.value as clientip, item.count as request_count 
+FROM ( 
+    SELECT unnest(approx_topk(clientip, 20)) as item 
+    FROM "demo1" 
+    ) 
+ORDER BY request_count 
+DESC
+```
+**Result of `approx_topk()` with `unnest()`**
+This provides a flat output as shown below: 
+
+```json
+{ "value": "192.168.1.100", "count": 2650 }
+{ "value": "10.0.0.5", "count": 2230 }
+{ "value": "203.0.113.50", "count": 2210 }
+...
+```
+
+---
+
+## `GROUP BY` versus `approx_topk`
+
+### How GROUP BY Works
+The traditional way to find the top values in a field is by using a `GROUP BY` query combined with `ORDER BY` and `LIMIT`. <br>
+    For example:
+
+    ```sql
+
+    SELECT clientip AS x_axis_1, COUNT(*) AS y_axis_1 
+    FROM cdn_production 
+    GROUP BY x_axis_1 
+    ORDER BY y_axis_1 DESC 
+    LIMIT 10
+    ```
+    This query counts how many times each unique `clientip` appears and returns the **top 10** based on that count.
+
+??? info "Why Traditional `GROUP BY` Breaks in Large Datasets"
+    In large datasets with high-cardinality fields, the query is executed across multiple querier nodes. Each node uses multiple CPU cores to process the data. The data is split into partitions, and each core handles a subset of partitions.
+
+    Consider the following scenario:
+
+    - Dataset contains `3 million` unique client IPs.
+    - Query runs using `60` CPU cores.
```

Comment:
this line should be runs using `60` querier nodes.

and each node have `60` cpu cores. each core will have one partition. so at the end is:

3m * 60(nodes) * 60(cores/partitions)